src/main/java/build/buf/protovalidate/ValidatorFactory.java

```diff
@@ -0,0 +1,159 @@
+import com.google.protobuf.Descriptors.Descriptor;
+import java.util.Collections;
+import java.util.List;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * ValidatorFactory is used to create a validator.
+ *
+ * <p>Validators can be created with an optional {@link Config} to customize behavior. They can also
+ * be created with a list of seed descriptors to warmup the validator cache ahead of time as well as
+ * an indicator to lazily-load any descriptors not provided into the cache.
+ */
+public class ValidatorFactory {
+  // Prevent instantiation
+  private ValidatorFactory() {}
+  ;
+
+  /**
+   * An eager builder behaves the same as a regular Builder, but attempts to warmup the validator
+   * cache with a given list of Descriptors.
+   */
+  public static class EagerBuilder extends BaseBuilder<EagerBuilder> {
+    private final List<Descriptor> descriptors;
+    private final boolean disableLazy;
+
+    EagerBuilder(List<Descriptor> descriptors, boolean disableLazy) {
+      this.descriptors = Collections.unmodifiableList(descriptors);
+      this.disableLazy = disableLazy;
+    }
+
+    /**
+     * Build the validator, warming up the cache with any provided descriptors.
+     *
+     * @return A Validator instance
+     * @throws CompilationException If any of the given descriptors' validation rules fail
+     *     processing while warming up the cache.
+     * @throws IllegalStateException If disableLazy is set to true and no descriptors are passed.
+     */
+    public Validator build() throws CompilationException, IllegalStateException {
+      if (disableLazy && this.descriptors.size() == 0) {
+        throw new IllegalStateException(
+            "a list of descriptors is required when disableLazy is true");
+      }
+
+      Config cfg = this.config;
+      if (cfg == null) {
+        cfg = Config.newBuilder().build();
+      }
+      return new ValidatorImpl(cfg, this.descriptors, this.disableLazy);
+    }
+
+    @Override
+    EagerBuilder self() {
+      return this;
+    }
+
+    boolean getDisableLazy() {
+      return this.disableLazy;
+    }
+
+    List<Descriptor> getDescriptors() {
+      return this.descriptors;
+    }
+  }
+
+  /** A builder class used for building a validator. */
+  public static class Builder extends BaseBuilder<Builder> {
+    // Default constructor
+    private Builder() {}
+
+    /**
+     * Build a new validator
+     *
+     * @return A new {@link Validator} instance.
+     */
+    public Validator build() {
+      Config cfg = this.config;
+      if (cfg == null) {
+        cfg = Config.newBuilder().build();
+      }
+      return new ValidatorImpl(cfg);
+    }
+
+    @Override
+    Builder self() {
+      return this;
+    }
+  }
+
+  /**
+   * A convenience function for creating a new validator with a default configuration.
+   *
+   * @return A Validator instance
+   */
+  public static Validator defaultInstance() {
+    return newBuilder().build();
```

Comment:
I think we should leave this out - it isn't that much more to type to have someone just do `ValidatorFactory.newBuilder().build()`. We could always add it back if desired.

This method as written feels like it is returning the same instance each time (similar to the global validator in pv-go: https://pkg.go.dev/buf.build/go/protovalidate#Validator), not a new instance each time w/ the default config.